src/Agent.Sdk/Util/ProcessUtil.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Management;
+
+namespace Microsoft.VisualStudio.Services.Agent.Util
+{
+    public static class WindowsProcessUtil
+    {
+        public static Process GetParentProcess(int processId)
+        {
+            try
+            {
+                using var query = new ManagementObjectSearcher($"SELECT * FROM Win32_Process WHERE ProcessId={processId}");
+                var process = query.Get().OfType<ManagementObject>().First();
+                var parentProcessId = (int)(uint)process["ParentProcessId"];
+                return Process.GetProcessById(parentProcessId);
+            }
+            catch
+            {
+                return null;
```

Comment:
Maybe it will be better to throw and catch exception instead of returning null?

Reply:
Done